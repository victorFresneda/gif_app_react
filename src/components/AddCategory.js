import React, { useState } from 'react'
import PropTypes  from 'prop-types'        

const AddCategory = ({setCategories}) => {

    const [inptValue, setinptValue] = useState('');

    const handleInputChange = (e) => {
        setinptValue(e.target.value);
        console.log('llamado necesario')
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if( inptValue.trim().length > 2) {
            setCategories( cats => [inptValue, ...cats,]);
            setinptValue('');
        }
        
        
    }

    return (
        
        <form onSubmit={ handleSubmit }>
            <input 
                placeholder="Que buscas?"
                type="text"
                value={ inptValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.prototypes = { 
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
