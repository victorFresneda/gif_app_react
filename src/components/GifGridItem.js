import React from 'react'

const GifGridItem = ({title, url}) => {

    

    return (
        <div id="mostrar-gifs">
            <img src={ url } alt={ title }/>
            <p> { title } </p>
        </div>
    )
}

export default GifGridItem
