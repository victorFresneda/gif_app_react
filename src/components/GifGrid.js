import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

const GifGrid = ( { categoria } ) => {

    const {data} = useFetchGifs( categoria);

    

    return (
        <>
            <h3 className="animate__animated animate__bounceInLeft"> { categoria } </h3>

            

            <div className="card-grid">


                    {
                        data.map( img => (
                            <GifGridItem
                                key={ img.id }
                                {...img}
                            />
                        ))
                    }

            </div>
        </>
    )
}

export default GifGrid
