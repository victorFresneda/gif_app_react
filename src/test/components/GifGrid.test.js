import React from "react"
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
import { GifGridItem } from "../../components/GifGridItem";
jest.mock( '../../hooks/useFetchGifs' )

describe('Pruebas en el <GifGrid />', () => {

    const categoria = 'One punch';

    test('debe mostrarse correctamnete ', () => {

        useFetchGifs.mockReturnValue({

            data: [],
            loading: true 

        });

        
        const wrapper = shallow( <GifGrid category ={ categoria } /> );
        expect( wrapper).toMatchSnapshot();  
        
    })

    test('Debe de mostrar items cuando se carga imagenes em useFetchGifs ', () => {
        

        const gifs = [{

            id: 'ABC',
            url:'https:/localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({

            data: gifs,
            loading: false 

        });


        const wrapper = shallow( <GifGrid category ={ categoria } /> );

        expect( wrapper.find('p').exists() ).toBe(false);
        expect(wrapper.find(GifGridItem).length).toBe( gifs.length ); 
        
    })
    
    
})
