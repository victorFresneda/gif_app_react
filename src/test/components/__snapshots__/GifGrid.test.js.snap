// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe mostrarse correctamnete  1`] = `
<div>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
     
     
  </h3>
  <div
    className="card-grid"
  />
</div>
`;
